server/sonar-server/src/main/java/org/sonar/server/measure/index/ProjectMeasuresIndex.java

```diff
@@ -245,38 +248,36 @@ private static AbstractAggregationBuilder createRangeFacet(String metricKey, Dou
         }
       });
 
-    return AggregationBuilders.nested("nested_" + metricKey)
-      .path(FIELD_MEASURES)
+    return AggregationBuilders.nested("nested_" + metricKey, FIELD_MEASURES)
       .subAggregation(
-        AggregationBuilders.filter("filter_" + metricKey)
-          .filter(termsQuery(FIELD_MEASURES_KEY, metricKey))
+        AggregationBuilders.filter("filter_" + metricKey, termsQuery(FIELD_MEASURES_KEY, metricKey))
           .subAggregation(rangeAgg));
   }
 
   private static AbstractAggregationBuilder createNoDataFacet(String metricKey) {
-    return AggregationBuilders.filter("no_data_" + metricKey)
-      .filter(boolQuery()
-        .mustNot(nestedQuery(FIELD_MEASURES, termQuery(FIELD_MEASURES_KEY, metricKey))));
+    return AggregationBuilders.filter(
+      "no_data_" + metricKey,
+      boolQuery().mustNot(nestedQuery(FIELD_MEASURES, termQuery(FIELD_MEASURES_KEY, metricKey), ScoreMode.Avg)));
   }
 
   private static AbstractAggregationBuilder createRatingFacet(String metricKey) {
-    return AggregationBuilders.nested("nested_" + metricKey)
-      .path(FIELD_MEASURES)
+    return AggregationBuilders.nested("nested_" + metricKey, FIELD_MEASURES)
       .subAggregation(
-        AggregationBuilders.filter("filter_" + metricKey)
-          .filter(termsQuery(FIELD_MEASURES_KEY, metricKey))
-          .subAggregation(filters(metricKey)
-            .filter("1", termQuery(FIELD_MEASURES_VALUE, 1d))
-            .filter("2", termQuery(FIELD_MEASURES_VALUE, 2d))
-            .filter("3", termQuery(FIELD_MEASURES_VALUE, 3d))
-            .filter("4", termQuery(FIELD_MEASURES_VALUE, 4d))
-            .filter("5", termQuery(FIELD_MEASURES_VALUE, 5d))));
+              AggregationBuilders.filter("filter_" + metricKey, termsQuery(FIELD_MEASURES_KEY, metricKey))
+                      .subAggregation(filters(metricKey,
+                              new KeyedFilter("1", termQuery(FIELD_MEASURES_VALUE, 1d)),
+                              new KeyedFilter("2", termQuery(FIELD_MEASURES_VALUE, 2d)),
+                              new KeyedFilter("3", termQuery(FIELD_MEASURES_VALUE, 3d)),
+                              new KeyedFilter("4", termQuery(FIELD_MEASURES_VALUE, 4d)),
+                              new KeyedFilter("5", termQuery(FIELD_MEASURES_VALUE, 5d)))));
   }
 
   private static AbstractAggregationBuilder createQualityGateFacet() {
-    FiltersAggregationBuilder qualityGateStatusFilter = AggregationBuilders.filters(ALERT_STATUS_KEY);
-    QUALITY_GATE_STATUS.entrySet().forEach(entry -> qualityGateStatusFilter.filter(entry.getKey(), termQuery(FIELD_QUALITY_GATE_STATUS, entry.getValue())));
-    return qualityGateStatusFilter;
+    return AggregationBuilders.filters(
+      ALERT_STATUS_KEY,
+      QUALITY_GATE_STATUS.entrySet().stream()
+        .map(entry -> new KeyedFilter(entry.getKey().toString(), termQuery(FIELD_QUALITY_GATE_STATUS, entry.getValue())))
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') "getKey" returns a string, there's no need to call "toString()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1858)